router/transformer/transformer_test.go

```diff
@@ -200,7 +226,7 @@ func TestProxyRequest(t *testing.T) {
 			assert.Equal(t, tc.ExpectedStCode, stCd)
 			if gjson.GetBytes([]byte(resp), "message").Raw != "" {
 				require.JSONEq(t, tc.ExpectedBody, resp)
-			} else if tc.ExpectedStCode == http.StatusGatewayTimeout {
+			} else if tc.ExpectedStCode == http.StatusGatewayTimeout || tc.ExpectedStCode == http.StatusNotFound {
```

Comment:
These test cases are getting complex with many if conditions and is it possible to implement them as separate test case instead of looping over them?

Reply:
I changed the test-cases a bit, can you check and let me know if you still feel they're complicated

Reply:
http.StatusInternalServerError is covered?

Reply:
I got the logic but it is taking time to understand see if you can simplify further.

Reply:
I would invert the logic of these tests, making the expectations more powerful.

```go
else if tc.ExpectedBody != "" {
				expectedBodyStr := fmt.Sprintf(tc.ExpectedBody, srv.URL)
				require.Equal(t, expectedBodyStr, resp)
}
```

If ExpectedBody is defined, we need to make the check, we don't want additional business logic on your testing here.

If there is some condition you don't care about or you don't expect a body just don't define it.

Same goes for the test above, which is quite dangerous:
```go
if gjson.GetBytes([]byte(resp), "message").Raw != "" {
				require.JSONEq(t, tc.ExpectedBody, resp)
			}
```
If you don't find the message in your response you are not doing the comparison, unexpected behaviour can slip in from that.

Reply:
I made a changes, can you once check and let me know if it feels good ?